sympy/integrals/tests/test_intpoly.py

```diff
@@ -0,0 +1,155 @@
+
+    hexagon = [((-1 / 2, -sqrt(3) / 2), 0),
+               ((-1, 0), sqrt(3) / 2),
+               ((-1 / 2, sqrt(3) / 2), sqrt(3)),
+               ((1 / 2, sqrt(3) / 2), sqrt(3)),
+               ((1, 0), sqrt(3) / 2),
+               ((1 / 2, -sqrt(3) / 2), 0)]
+    assert polytope_integrate(hexagon, 1) == S(3*sqrt(3)) / 2
+
+    #  Non-convex polytopes
+    #  Vertex representation
+    assert polytope_integrate(Polygon(Point(-1, -1), Point(-1, 1),
+                                      Point(1, 1), Point(0, 0),
+                                      Point(1, -1)), 1) == 3
+    assert polytope_integrate(Polygon(Point(-1, -1), Point(-1, 1),
+                                      Point(0, 0), Point(1, 1),
+                                      Point(1, -1), Point(0, 0)), 1) == 2
+    #  Hyperplane representation
+    assert polytope_integrate([((-1, 0), 1), ((0, 1), 1), ((1, -1), 0),
+                               ((1, 1), 0), ((0, -1), 1)], 1) == 3
+    assert polytope_integrate([((-1, 0), 1), ((1, 1), 0), ((-1, 1), 0),
+                               ((1, 0), 1), ((-1, -1), 0),
+                               ((1, -1), 0)], 1) == 2
+
+    #  Tests for 2D polytopes mentioned in Chin et al(Page 10):
+    #  http://dilbert.engr.ucdavis.edu/~suku/quadrature/cls-integration.pdf
+    fig1 = Polygon(Point(1.220, -0.827), Point(-1.490, -4.503),
+                   Point(-3.766, -1.622), Point(-4.240, -0.091),
+                   Point(-3.160, 4), Point(-0.981, 4.447),
+                   Point(0.132, 4.027))
+    assert polytope_integrate(fig1, x**2 + x*y + y**2) ==\
+        S(2031627344735367)/(8*10**12)
+
+    fig2 = Polygon(Point(4.561, 2.317), Point(1.491, -1.315),
+                   Point(-3.310, -3.164), Point(-4.845, -3.110),
+                   Point(-4.569, 1.867))
+    assert polytope_integrate(fig2, x**2 + x*y + y**2) ==\
+        S(517091313866043)/(16*10**11)
+
+    fig3 = Polygon(Point(-2.740, -1.888), Point(-3.292, 4.233),
+                   Point(-2.723, -0.697), Point(-0.643, -3.151))
+    assert polytope_integrate(fig3, x**2 + x*y + y**2) ==\
+        S(147449361647041)/(8*10**12)
+
+    fig4 = Polygon(Point(0.211, -4.622), Point(-2.684, 3.851),
+                   Point(0.468, 4.879), Point(4.630, -1.325),
+                   Point(-0.411, -1.044))
+    assert polytope_integrate(fig4, x**2 + x*y + y**2) ==\
+        S(180742845225803)/(10**12)
+
+    #  Tests for many polynomials with maximum degree given.
+    tri = Polygon(Point(0, 3), Point(5, 3), Point(1, 1))
+    polys = []
+    expr1 = x**9*y + x**7*y**3 + 2*x**2*y**8
+    expr2 = x**6*y**4 + x**5*y**5 + 2*y**10
+    expr3 = x**10 + x**9*y + x**8*y**2 + x**5*y**5
+    polys.extend((expr1, expr2, expr3))
+    result_dict = polytope_integrate(tri, polys, max_degree=10)
+    assert result_dict[expr1] == 615780107/594
+    assert result_dict[expr2] == 13062161/27
+    assert result_dict[expr3] == 1946257153/924
+
+@XFAIL
```

Comment:
I mean can you comment into the source code why it fails. Is it because these polynomials are interesecting? If so, then just put something like this into the code:
```python
# This test fails because the polynomials are interesecting and the code cannot handle it yet. This will be fixed in the future.
```
or something like that.